licenses/models.py

```diff
@@ -9,88 +9,37 @@
 Some licenses ahve a dcq:isReplacedBy element.
 
 """
-from django.conf import settings
 from django.db import models
 from django.urls import reverse
 from django.utils import translation
-from django.utils.translation import ugettext
 
-from i18n import DEFAULT_LANGUAGE_CODE
+from i18n import DEFAULT_LANGUAGE_CODE, DEFAULT_JURISDICTION_LANGUAGES
 from licenses import (
     FREEDOM_LEVEL_MIN,
     FREEDOM_LEVEL_MID,
     FREEDOM_LEVEL_MAX,
 )
 
 
-class Creator(models.Model):
-    url = models.URLField(
-        unique=True, max_length=200, help_text="E.g. http://creativecommons.org"
-    )
-
-    def __str__(self):
-        return f"Creator<{self.url}>"
-
-
-class Jurisdiction(models.Model):
-    code = models.CharField(max_length=9, unique=True, blank=False)
-
-    @property
-    def about(self):
-        # Not using "https:" deliberately because this is the "official"
-        # name of the jurisdiction in the RDF.
-        return f"http://creativecommons.org/international/{self.code}/"
-
-    default_language = models.ForeignKey(
-        "Language", null=True, on_delete=models.CASCADE
-    )
-
-    def __str__(self):
-        return f"Jurisdiction<{self.code}>"
-
-
-class LicenseClass(models.Model):
-    # <cc:licenseClass rdf:resource="http://creativecommons.org/license/"/>
+class LegalCode(models.Model):
     url = models.URLField(
-        unique=True,
         max_length=200,
-        help_text="E.g. http://creativecommons.org/license/",
+        help_text="E.g. http://creativecommons.org/licenses/by-nd/3.0/rs/legalcode.sr-Cyrl",
     )
-
-    def __str__(self):
-        return f"LicenseClass<{self.url}>"
-
-
-class Language(models.Model):
-    code = models.CharField(
-        unique=True,
+    license = models.ForeignKey("licenses.License", on_delete=models.CASCADE, related_name="legal_codes")
+    language_code = models.CharField(
         max_length=7,
         help_text="E.g. 'en', 'en-ca', 'sr-Latn', or 'x-i18n'. Case-sensitive?",
     )
 
-    def __str__(self):
-        return f"Language<{self.code}>"
-
-    def name(self):
-        """Name of language (translated to current active language)"""
-        name = dict(settings.LANGUAGES)[self.code]
-        return ugettext(name)
-
-
-class LegalCode(models.Model):
-    url = models.URLField(
-        max_length=200,
-        help_text="E.g. http://creativecommons.org/licenses/by-nd/3.0/rs/legalcode.sr-Cyrl",
-    )
-    language = models.ForeignKey(Language, on_delete=models.CASCADE)
-
     class Meta:
         unique_together = [
-            ("url", "language"),
+            ("url", "language_code"),
+            ("license", "language_code"),
         ]
 
     def __str__(self):
-        return f"LegalCode<{self.language}, {self.url}>"
+        return f"LegalCode<{self.language_code}, {self.url}>"
 
 
 class License(models.Model):
```

Comment:
https://cc-vocabulary.netlify.app/?path=/docs/tokens-icons--figma

Can we store the cc licenses icon classes on the `Licenses` model?